codice/ddf
# DDF-4133 Improve security and communication with Solr server

#### What does this PR do?

1. Changes the DDF Solr connection from two-way TLS to one-way TLS
2. Adds basic authentication for the DDF to prove its identity to Solr
3. Adds a module to generate a basic auth password, change the password in Solr, encrypt the password, and save the encrypted password to a file.
4. Introduces new system properties to control behavior (see testing section)

#### Who is reviewing it? 
@paouelle 
@mcalcote 
@stustison 
#### Select relevant component teams: 
@codice/security 
@codice/solr 

#### Ask 2 committers to review/merge the PR and tag them here.
@ricklarsen - Documentation
@stustison
@vinamartin

#### How should this be tested?
1. Verification
   1. Review documentation changes in section _7.5.1.2.2. Solr Server Password Management_ of the documentation. Perform the configuration task described in the section. Confirm that the system behaves as documented.
  2. Change the values or combination of values of the new system properties and determine if they work as described in the documentation section _Managed Solr Server Properties_
solr.useBasicAuth
solr.username
solr.password
solr.attemptAutoPasswordChange

3. Validation
  1. Do the configuration options make sense?
  1. Does the design follow the "principle of least surprise"
  1. Are these things appropriately named? Are the names consistent and what you expected?
     1. Group ID, artifact ID
     2. Directory of Maven module
     3. Package names, class names
     4. Bundle names, feature names and descriptions?


#### Any background context you want to provide?
#### What are the relevant tickets?
DDF-4133

#### Screenshots
![screen shot 2018-11-28 at 3 30 26 pm](https://user-images.githubusercontent.com/133284/49186704-93dd6d80-f322-11e8-86f2-44522b2a7e6e.png)

#### Checklist:
- [X ] Documentation Updated

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
